upup/pkg/fi/cloudup/awstasks/tags.go

```diff
@@ -62,3 +62,15 @@ func intersectTags(tags []*ec2.Tag, desired map[string]string) map[string]string
 	}
 	return actual
 }
+
+func mapToTagSpecification(tags map[string]string, resourceType string) *ec2.TagSpecification {
```

Comment:
I think there are a few other places where TagSpecification is used and this should be used to replace the tag conversion there.